### PR TITLE
Make Routes Symmetric

### PR DESCRIPTION
**Pull request by harshkpatel**
## Proposed Changes

_(Describe your changes here. Also describe the motivation for your changes: what problem do they solve, or how do they improve the application or codebase? If this pull request fixes an open issue, [use a keyword to link this pull request to the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).)_

In the function ```strictRoutes```, some of the tuples have parameters which are looked up and passed into the controller function. The goal of this PR is to modify these tuples by moving these lookup parameters into the functions themselves to create a symmetry between all the elements in ```strictRoutes```.

<details>
<summary>Screenshots of your changes (if applicable)</summary>

</details>

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |          |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  |          |
| 🎨 _User interface change_ (change to user interface; provide screenshots)              |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |  x   |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

- [x] I have performed a self-review of my changes.
  - Check that all changed files included in this pull request are intentional changes.
  - Check that all changes are relevant to the purpose of this pull request, as described above.
- [ ] I have added tests for my changes, if applicable.
  - This is **required** for all bug fixes and new features.
- [ ] I have updated the project documentation, if applicable.
  - This is **required** for new features.
- [ ] If this is my first contribution, I have added myself to the list of contributors.
- [x] I have updated the project Changelog (this is required for all changes).

After opening your pull request:

- [x] I have verified that the CircleCI checks have passed.
- [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments

_(Include any questions or comments you have regarding your changes.)_

### CODE REVIEW

**Review by david-yz-liu — COMMENTED** [review]
@harshkpatel this is great. My only comment, throughout, is that I'd prefer using do notation instead of `(>>=)`. I suspect that the former will be a bit easier for new contributors to pick up, even though I'd still want them learning about monads in general!

**Comment by harshkpatel** [comment]
Changed everything to do notation!

File: app/Controllers/Course.hs [hunk 1]
```diff
@@ -8,14 +8,16 @@ import Data.List (sort, nub)
 import Database.Persist (Entity)
 import Database.Persist.Sqlite (SqlPersistM, runSqlite, selectList, entityVal)
 import Database.Tables as Tables (Courses, coursesCode)
-import Happstack.Server.SimpleHTTP (ServerPart, Response, toResponse)
+import Happstack.Server(lookText', ServerPart, Response, toResponse)
```
Comment (david-yz-liu) [inline app/Controllers/Course.hs:11]: insert a space before the `(`

File: app/Database/CourseInsertion.hs [hunk 2]
```diff
@@ -12,23 +12,25 @@ module Database.CourseInsertion
...
+import Control.Monad.IO.Class (liftIO)
 import qualified Data.Aeson as Aeson
-import qualified Data.ByteString.Lazy.Char8 as BSL
 import qualified Data.Text as T
 import Database.Persist.Class (selectKeysList)
 import Database.Persist.Sqlite (SqlPersistM, insert, insertMany_, insert_, runSqlite, selectFirst,
                                 (==.))
 import Database.Tables hiding (breadth, distribution, paths, shapes, texts)
-import Happstack.Server.SimpleHTTP (Response, toResponse)
+import Happstack.Server(lookBS, lookText', ServerPart, Response, toResponse)
```
Comment (david-yz-liu) [inline app/Database/CourseInsertion.hs:22]: insert a space before the `(`

File: app/Database/CourseQueries.hs [hunk 3]
```diff
@@ -144,8 +144,10 @@ buildMeetTimes meet = do
 
 -- | Looks up a graph using its title then gets the Shape, Text and Path elements
 -- for rendering graph (returned as JSON).
-getGraphJSON :: T.Text -> IO Response
-getGraphJSON graphName = getGraph graphName >>= withDefault
+getGraphJSON :: ServerPart Response
+getGraphJSON = do
+    graphName <- lookText' "graphName" 
+    liftIO $ getGraph graphName >>= withDefault
```
Comment (david-yz-liu) [inline app/Database/CourseQueries.hs:150]: Now that you've introduced using `do` notation, please change the `>>=` in here as well

File: app/Response/Calendar.hs [hunk 4]
```diff
@@ -12,12 +12,14 @@ import Data.Time.Calendar.OrdinalDate (fromMondayStartWeek, mondayStartWeek)
...
-import Happstack.Server (Response, ServerPart, toResponse)
+import Happstack.Server (Response, ServerPart, toResponse, look)
 import Text.Read (readMaybe)
 
 -- | Returns an ICS file of events as requested by the user.
-calendarResponse :: String -> ServerPart Response
-calendarResponse = liftIO . getCalendar . T.pack
+calendarResponse :: ServerPart Response
+calendarResponse = do
+    courses <- look "courses"
```
Comment (david-yz-liu) [inline app/Response/Calendar.hs:21]: Now that everything's combined in this function, take this opportunity to simplify a bit here by using `lookText'`

File: app/Response/Export.hs [hunk 5]
```diff
@@ -14,14 +14,18 @@ import Response.Image (returnImageData)
 import System.Directory (removeFile)
 
 -- | Returns an image of the timetable requested by the user.
-exportTimetableImageResponse :: T.Text -> String -> ServerPart Response
-exportTimetableImageResponse session selectedCourses = do
+exportTimetableImageResponse :: ServerPart Response
+exportTimetableImageResponse = do
+    session <- lookText' "session"
+    selectedCourses <- look "courses"
```
Comment (david-yz-liu) [inline app/Response/Export.hs:20]: As above, use `lookText'` instead of `look` to simplify

File: app/Response/Export.hs [hunk 6]
```diff
@@ -14,14 +14,18 @@ import Response.Image (returnImageData)
...
     (svgFilename, imageFilename) <- liftIO $ getActiveTimetable (T.pack selectedCourses) session
     liftIO $ returnImageData svgFilename imageFilename
 
 -- | Returns a PDF containing graph and timetable requested by the user.
-exportTimetablePDFResponse :: String -> String -> ServerPart Response
-exportTimetablePDFResponse selectedCourses graphInfo = do
+exportTimetablePDFResponse :: ServerPart Response
+exportTimetablePDFResponse = do
+    selectedCourses <- look "courses"
```
Comment (david-yz-liu) [inline app/Response/Export.hs:27]: Same here (`lookText'`)

File: app/Database/CourseQueries.hs [hunk 7]
```diff
@@ -140,7 +140,6 @@ buildMeetTimes meet = do
     return $ Tables.MeetTime' (entityVal meet) parsedTime
 
 -- ** Other queries
-
```
Comment (david-yz-liu) [inline app/Database/CourseQueries.hs:143]: Revert this change

File: app/Controllers/Timetable.hs [hunk 8]
```diff
@@ -74,10 +78,11 @@ returnPDF graphSvg graphImg fallTimetableSvg fallTimetableImg springTimetableSvg
     mapM_ removeFile [graphSvg, graphImg, fallTimetableSvg, fallTimetableImg, springTimetableSvg, springTimetableImg]
     return pdfName
 
-
```
Comment (david-yz-liu) [inline app/Controllers/Timetable.hs:77]: Revert this change

File: app/Controllers/Graph.hs [hunk 9]
```diff
@@ -1,15 +1,13 @@
...
+import Happstack.Server (ServerPart, Response, toResponse, ok, lookText', look)
 import MasterTemplate (masterTemplate, header)
 import Scripts (graphScripts)
 import Text.Blaze ((!))
 import qualified Text.Blaze.Html5 as H
 import qualified Text.Blaze.Html5.Attributes as A
-import qualified Data.Text as T (Text)
 import Control.Monad.IO.Class (liftIO)
 import Data.Aeson (object, (.=))
-
```
Comment (david-yz-liu) [inline app/Controllers/Graph.hs:12]: Revert this change (and the other blank line deletion changes in this file)

**Review by david-yz-liu — COMMENTED** [review]
@harshkpatel I left a few more inline comments; please also remove the `config.yaml` file from this commit (I assume you were testing with this, but please be careful with which files you add to commits).

**Comment by harshkpatel** [comment]
Just a quick question about the spacing, why do some functions have two lines of spacing between them and others only have one?

**Comment by david-yz-liu** [comment]
This could be an inconsistency, but for stylistic issues we can fix them separately (ideally with a linter like HLint). We shouldn't change style within other PRs.

**Review by david-yz-liu — APPROVED** [review]
